src/models/edge.ts

```diff
@@ -25,6 +26,18 @@ export interface IEdgePosition {
   target: any;
 }
 
+export enum EdgeLineStyleType {
+  SOLID = "solid",
+  DASHED = "dashed",
+  DOTTED = "dotted",
+  CUSTOM = "custom",
+}
+
+export interface IEdgeLineStyle {
+  type: EdgeLineStyleType;
+  dashPattern: number[];
+}
```

Comment:
This should be fixed from a typescript side to be more precise. For example, currently, you can have `{ type: "solid", dashPattern: [1, 2, 3] }` which is not correct. An array can only be used with `custom` type. E.g.

```typescript
export type IEdgeLineStyle =
  | { type: EdgeLineStyleType.SOLID }
  | { type: EdgeLineStyleType.DASHED }
  | { type: EdgeLineStyleType.DOTTED }
  | { type: EdgeLineStyleType.CUSTOM, pattern: number[] };
```

or even

```typescript
export type IEdgeLineStyle =
  | EdgeLineStyleType.SOLID
  | EdgeLineStyleType.DASHED
  | EdgeLineStyleType.DOTTED
  | { type: EdgeLineStyleType.CUSTOM, pattern: number[] };
```

Maybe the second one might even work better if users will use SOLID, DASHED or DOTTED more. The first one is easier to use from library point of view, the second one from user point of view.

Reply:
Nitpicking but I am also thinking about renaming `dashPattern` to just `pattern` because you already have a context which is a line style. What do you think? Also, what do you think about those two proposals above, which one makes more sense to you?

Reply:
I agree with your suggestion to fix the TypeScript type for `IEdgeLineStyle`.

Yes The second approach might even work better if users will use SOLID, DASHED or DOTTED more. However, I am thinking to use the first approach because it gives users a more consistent and predictable experience. With the first approach, users know that they can always specify the line style using the same syntax, regardless of whether they are using a predefined style or a custom style.

Reply:
Yep, agree with you. The first approach is slightly a better option because of the usage predictability.